### PR TITLE
Add test for requires_process_group_agent decorator

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27879 Add test for requires_process_group_agent decorator**

Differential Revision: [D17924096](https://our.internmc.facebook.com/intern/diff/D17924096)

### CODE REVIEW

**Comment by mrshenli** [comment]
Don't merge before #27789

**Review by xush6528 — COMMENTED** [review]
Lint errors?

**Comment by mrshenli** [comment]
@xush6528 yes, will rebase and fix lint after #27789 is merged.

**Review by xush6528 — APPROVED** [review]

**Pull request closed without merging**
